src/redux-middleware/__tests__/pushQueue.ts

```diff
@@ -0,0 +1,96 @@
+import { store } from '../../store'
+import { HOME_PATH } from '../../constants'
+import { clear, importText, existingThoughtChange } from '../../action-creators'
+import { initialize } from '../../initialize'
+import { getThought as getThoughtSelector } from '../../selectors'
+import * as dexie from '../../data-providers/dexie'
+import getThought from '../../data-providers/data-helpers/getThought'
+import { DataProvider } from '../../data-providers/DataProvider'
+import testTimer from '../../test-helpers/testTimer'
+import { SimplePath } from '../../types'
+
+/*
+  Note: sinon js fake timer is used to overcome some short comming we have with jest's fake timer.
+  For details: https://github.com/cybersemics/em/issues/919#issuecomment-739135971
+*/
+
+const fakeTimer = testTimer()
+
+const db = dexie as DataProvider
+
+beforeEach(async () => {
+  fakeTimer.useFakeTimer()
+  initialize()
+  await fakeTimer.runAllAsync()
+  fakeTimer.useRealTimer()
+})
```

Comment:
It doesn't need to test components. But it would be alright to use `createTestApp` instead of using `initialize` and  fake timers here. I will change it.